django/db/migrations/operations/base.py

```diff
@@ -138,6 +142,35 @@ def reduce(self, operation, app_label):
             return [self]
         return False
 
+    def check(self):
+        if self.system_check_removed_details is not None:
+            return [
+                checks.Error(
+                    self.system_check_removed_details.get(
+                        "msg",
+                        "%s has been removed except for support in historical "
+                        "migrations." % self.__class__.__name__,
+                    ),
+                    hint=self.system_check_removed_details.get("hint"),
+                    obj=self,
+                    id=self.system_check_removed_details.get("id", "migrations.EXXX"),
+                )
+            ]
+        elif self.system_check_deprecated_details is not None:
+            return [
+                checks.Warning(
+                    self.system_check_deprecated_details.get(
+                        "msg", "%s has been deprecated." % self.__class__.__name__
+                    ),
+                    hint=self.system_check_deprecated_details.get("hint"),
+                    obj=self,
+                    id=self.system_check_deprecated_details.get(
+                        "id", "migrations.WXXX"
+                    ),
+                )
+            ]
+        return []
```

Comment:
This is exactly the same as for `models.Field`. What do you think about moving this logic to a mixin (`django.utils.deprecation`) and using it in both places :thinking: ? e.g.
```python
class DeprecationForHistoricalMigration(Mixin):
    system_check_deprecated_details = None
    system_check_removed_details = None

    def check(self):
        if self.system_check_removed_details is not None:
            return [
                checks.Error(
                    self.system_check_removed_details.get(
                        "msg",
                        "%s has been removed except for support in historical "
                        "migrations." % self.__class__.__name__,
                    ),
                    hint=self.system_check_removed_details.get("hint"),
                    obj=self,
                    id=self.system_check_removed_details.get("id", "migrations.EXXX"),
                )
            ]
        elif self.system_check_deprecated_details is not None:
            return [
                checks.Warning(
                    self.system_check_deprecated_details.get(
                        "msg", "%s has been deprecated." % self.__class__.__name__
                    ),
                    hint=self.system_check_deprecated_details.get("hint"),
                    obj=self,
                    id=self.system_check_deprecated_details.get(
                        "id", "migrations.WXXX"
                    ),
                )
            ]
        return []
```